### ISSUE TITLE
CLI error: "Dongle needs to be reinitialized. The SIM card is not ready yet"

### ISSUE THREAD

**Issue by VGBLTSK** [bg111/asterisk-chan-dongle#236]
Guys, 
Probably someone can help with any advice on the issue. Asterisk throws ‘Dongle needs to be reinitialized. The SIM card is not ready yet’ message for one of my sim cards. I have three e171 dongles, two of them work perfectly. I tried to swap sim cards and ascertained that the problem is sim-related. After poking around with chan_dongle code I received evidence that the problem has something to do with AT+CNUM command, being used by chan_dongle module upon initializing phase. I pinged the dongle with that sim card with socat tool and got the following responses:
at+cpbs?

+CPBS: "ON",1,5

OK

AT+CNUM

ERROR

That makes me to believe that every time chan_dongle attempts to initialize the dongle it palls it with CMD_AT_CNUM command and fails while getting ERROR response. 
The own number (MSISDN) of the sim card seems impossible to be changed using AT commands not directly while incerted into a cell phone (it can only be read by device). AT+CPBS=”ON” and AT+CPBW=,"+XXXXXXXXXXX",128,"Voice" gave me just ERROR
I have got stuck…. Replaced the sim card but still no joy (the same story)– it doesn’t seem to be provided by my mobile operator. Is that query important or can be commented out in a source code and re-compiled for running Asterisk inside Raspbx distro?

**Comment by abdoufma** [comment]
I believe I have the exact same issue: one of my SIM cards doesn't work with any of my dongles (same error message) did you manage to solve your issue?

**Comment by VGBLTSK** [comment]
Yes, all I had to do is to manually set a sim card phone number.

**Comment by abdoufma** [comment]
You mean using  AT+CPBW ? 
Why does THIS SIM in particular require a number, and not all the other ones?  especially that it used to work perfectly fine before this issue happened... Do you have any explanation for that? (Sorry, I'm new to this)

Thanks for the reply by the way!

**Comment by VGBLTSK** [comment]
Just to make sure you're using the right syntax:
AT+CPBS="ON"
AT+CPBW=1,"8885551212",,"Voice" ; (yes, without number type)

In my particular case it was due to my carrier who used to manufacture and issue sim cards without numbers stored on SIM. This manual step brought a salvation. You can try and pitch chan_dongle code at your needs. Since it is open you may disable that verification at all.  But storing a number on SIM is a lot easier. 

Evgenii.

**Comment by abdoufma** [comment]
Thanks again for the reply! I’ll try that tomorrow when I’m at work.

**Comment by VGBLTSK** [comment]
You are welcome.

**Comment by mihai9621** [comment]
Hello, please help me, I can't save the number, what should I do?

AT+CPBS="ON"
OK
AT+CPBS?
+CPBS: "ON",4,4
AT+CPBW=?
+CPBW: (1-4),24,(128-255),16
AT+CPBW=1,"8885551212",,"Voice"
ERROR
